### PR TITLE
feat(kuma-cp) Generate cert with multiple SAN URIs

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

This is the second approach to solve a problem in this PR https://github.com/Kong/kuma/pull/757
We could not go that way, because you [cannot](https://github.com/envoyproxy/envoy/blob/74290ef76a76fbbf50f072dc33438791f93f68c7/api/envoy/api/v2/auth/tls.proto#L122) pass multiple SDS sources for many DP certs for outbound.

Instead, we can generate one certificate for DP with multiple URI SANs.

### CODE REVIEW

File: pkg/core/ca/issuer/issuer.go [hunk 1]
```diff
@@ -42,18 +42,22 @@ func NewWorkloadCert(ca util_tls.KeyPair, mesh string, workload string, certOpts
...
 	}
 	return util_tls.ToKeyPair(workloadKey, workloadCert)
 }
 
-func newWorkloadCert(signer crypto.PrivateKey, parent *x509.Certificate, trustDomain string, workload string, publicKey crypto.PublicKey, certOpts ...CertOptsFn) ([]byte, error) {
-	spiffeID := &url.URL{
-		Scheme: "spiffe",
-		Host:   trustDomain,
-		Path:   workload,
+func newWorkloadCert(signer crypto.PrivateKey, parent *x509.Certificate, trustDomain string, services []string, publicKey crypto.PublicKey, certOpts ...CertOptsFn) ([]byte, error) {
```
Comment (nickolaev) [inline pkg/core/ca/issuer/issuer.go:52]: That's a lot of arguments to pass. Maybe a structure could be better.

**Comment by jakubdyszkiewicz** [comment]
> The current implementation looks good. Is it theoretically possible to generate own clusters for every inbound, so the overall amount of clusters is inbound*services? Probably that would allow us to have a separate certificate for each inbound. Or that doesn't make sense?

Every inbound has own cluster right now. The problem is that for outbound cluster you need to present those certs to client and you can only present one cert, therefore we need one cert with multiple SANs

**Comment by jakubdyszkiewicz** [comment]
and also, since we need to present all certs to outgoing connection, to be honest, it makes more sense to have all IDs in one cert.

File: pkg/core/ca/issuer/issuer.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/core/ca/issuer/issuer.go]: Ok, I applied some refactoring in this area. It doesn't make sense to carry all of those args across 2 methods.

**Pull request merged**